### PR TITLE
M2-2870 Score report common fields validation

### PR DESCRIPTION
**Pull request by wildkiomi**

### CODE REVIEW

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx [hunk 1]
```diff
@@ -785,6 +785,20 @@ export const getAppletTabs = ({
   },
 ];
 
+export const testIsReportCommonFieldsRequired = (
+  isScoreReport: boolean,
+  value: boolean,
+  context: unknown,
+) => {
+  if (isScoreReport) {
+    const conditionalLogic = get(context, 'from.0.value.conditionalLogic')?.length;
```
Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:794]:
```suggestion
    const hasConditionalLogic = get(context, 'from.0.value.conditionalLogic')?.length;
```

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx [hunk 2]
```diff
@@ -785,6 +785,20 @@ export const getAppletTabs = ({
...
+
+    return !!conditionalLogic || value;
```
Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:796]: is the double exclamation mark needed here?
as I see above, conditional logic can be undefined, 0, or other number

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx [hunk 3]
```diff
@@ -785,6 +785,20 @@ export const getAppletTabs = ({
   },
 ];
 
+export const testIsReportCommonFieldsRequired = (
+  isScoreReport: boolean,
+  value: boolean,
```
Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:790]: could you pls give more understandable name for this parameter?

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx (hunk 2 above)
Comment (wildkiomi) [reply src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:796]: this function should return boolean. so !! is required

**Review by IhorNesterenko99 — APPROVED** [review]

**Review by muhurov — APPROVED** [review]

**Pull request merged**